src/box/sql/func.c

```diff
@@ -992,8 +992,11 @@ func_round(struct sql_context *ctx, int argc, const struct Mem *argv)
 
 	double d = argv[0].u.r;
 	struct Mem *res = ctx->pOut;
-	if (n != 0)
-		return mem_set_double(res, atof(tt_sprintf("%.*f", n, d)));
+	if (n != 0) {
+		int precision = MIN(n, INT_MAX);
```

Comment:
Can you clarify how the precision may be so large?

Reply:
+1. This looks confusing. A simple cast wouldn't raise this question, I guess.

Reply:
Precision can be any `uint64_t`. Because that is what stored in `struct Mem`. Hence need to properly check for the entire range. `printf`'s format `.*` takes an int - need to pass an int. As for a cast, it won't work - it will take low 32 bits instead of doing what I do in this patch. Consider this example:
```C
#include <stdio.h>
#include <stdint.h>

int
main(void)
{
	uint64_t x = 0xff00000000000000;
	int x2 = (int)x;
	printf("%x\n", x2);
	return 0;
}
```
It will print `0`, not `INT_MAX`.
It is questionable whether such a huge precision should even be allowed, but it is out of scope of the patch. I already created a ticket to investigate the strange behaviour: https://github.com/tarantool/tarantool/issues/6650.

Reply:
Okay, makes sense.